includes/GlobalNewFilesHooks.php

```diff
@@ -8,49 +8,24 @@ public static function onCreateWikiTables( &$tables ) {
 	}
 
 	public static function onUploadComplete( $uploadBase ) {
-		$config = MediaWikiServices::getInstance()->getConfigFactory()->makeConfig( 'globalnewfiles' );
-
-		$uploadedFile = $uploadBase->getLocalFile();
-
-		$dbw = self::getGlobalDB( DB_PRIMARY );
-
-		$wiki = new RemoteWiki( $config->get( 'DBname' ) );
-
-		$dbw->insert(
-			'gnf_files',
-			[
-				'files_dbname' => $config->get( 'DBname' ),
-				'files_name' => $uploadedFile->getName(),
-				'files_page' => $config->get( 'Server' ) . $uploadedFile->getDescriptionUrl(),
-				'files_private' => (int)$wiki->isPrivate(),
-				'files_timestamp' => $dbw->timestamp(),
-				'files_url' => $uploadedFile->getViewURL(),
-				'files_user' => $uploadedFile->getUser()
-			],
-			__METHOD__
+		JobQueueGroup::singleton()->push(
+			new GlobalNewFilesInsertJob( [ 'uploadbase' => $uploadBase ] )
 		);
 	}
 
-	/**
-	 * Hook to FileDeleteComplete
-	 * @param File $file
-	 * @param File $oldimage
-	 * @param Article $article
-	 * @param User $user
-	 * @param string $reason
-	 */
 	public static function onFileDeleteComplete( $file, $oldimage, $article, $user, $reason ) {
-		$config = MediaWikiServices::getInstance()->getConfigFactory()->makeConfig( 'globalnewfiles' );
+		JobQueueGroup::singleton()->push(
+			new GlobalNewFilesDeleteJob( [ 'file' => $file ] )
+		);
+	}
 
-		$dbw = self::getGlobalDB( DB_PRIMARY );
+	public static function onTitleMoveComplete( $title, $newTitle, $user, $oldid, $newid, $reason, $revision ) {
```

Comment:
We should migrate this hook to PageMoveComplete (TitleMoveComplete was deprecated in 1.35). See [manual](https://www.mediawiki.org/wiki/Manual:Hooks/TitleMoveComplete). Also, there's no problem with having a return value in this function, but doesn't it make more sense to just do 
```php
if ( $title->inNamespace( NS_FILE ) ) {
	JobQueueGroup::singleton()->push(
		new GlobalNewFilesMoveJob( [ 'title' => $title, 'newtitle' => $newTitle ] )
	);
}
```

Reply:
I noticed the hook was deprecated but planned to do it in a follow-up, but I can do it in here if you think that'd be a better idea?

As for check, that's true can do that.

Reply:
Sure, we can do that in a follow up, especially since it isn't just a plug and swap replacement.

Reply:
Sounds good then. I redid the check now.